test/integration/xfcc_integration_test.cc

```diff
@@ -0,0 +1,243 @@
+#include "xfcc_integration_test.h"
```

Comment:
It seems overkill to create an entire new test config file for every feature we want to integrate. I think this is a general issue with integration tests today - there's no easy way to templatize the creation of the configs to turns on/off individual features while retaining a common base. This results in a lot of boiler plate to maintain. @mattklein123 for comment.